src/UnivariateFeatureSelector.jl

```diff
@@ -0,0 +1,193 @@
+"""
+`UnivariateFeatureSelector` has the following fields:
+
+* `k::Union{Int64,Nothing}` - Select top `k` features with the highest correlation to target
+  variable. You could ignore this by specifying k == nothing. This defaults to nothing.
+* `threshold::Union{Float64,Nothing}` - Select features with correlation more than or equal to
+  threshold. To ignore, simply set threshold to nothing (default behavior).
+"""
+mutable struct UnivariateFeatureSelector
+    k::Union{Int64,Nothing}
+    threshold::Union{Float64,Nothing}
+end
+
+function UnivariateFeatureSelector(;
+    k::Union{Int64,Nothing} = nothing,
+    threshold::Union{Float64,Nothing} = nothing,
+)
+    UnivariateFeatureSelector(k, threshold)
+end
+
+"""
+    function select_features(selector,
+                             X::DataFrame,
+                             y::Vector;
+                             verbose::Bool=false,
+                             return_val::Bool=false)
+
+Select features based on the importance, which is defined by `method` to
+target `y`. if `verbose` is true, logs will be printed - this defaults to
+false. If `return_val` is false, this function will return only the feature
+feature names, otherwise, tuple of selected feature names and the
+correlation value are returned.
+
+The `method` you choose will determine the scoring, below is the scoring
+with available statistical method to obtain them.
+
+    * Correlation - higher score means more important
+
+        * `pearson_correlation`
+
+    * P-value - lower score means more important
+
+        * `f_test`
+        * `chisq_test`
+
+If you have feature `X_data` as matrix and feature names `X_features` as a
+Vector, you can replace `X` with `X_data` and `X_features` (in this order).
+
+# Example
+
+```jldoctest
+julia> using RDatasets, FeatureSelector, DataFrames
+
+julia> boston = dataset("MASS", "Boston");
+
+julia> selector = UnivariateFeatureSelector(k=5)
+UnivariateFeatureSelector(5, nothing)
+
+julia> select_features(
+           selector,
+           pearson_correlation,
+           boston[:, Not(:MedV)],
+           boston.MedV
+       )
+5-element Array{Symbol,1}:
+ :LStat
+ :Rm
+ :PTRatio
+ :Indus
+ :Tax
+
+```
+"""
+function select_features(
+    selector::UnivariateFeatureSelector,
+    method::Function,
```

Comment:
Actually I was thinking that `method` would be a field of `UnivariateFeatureSelector`.

Reply:
Ahh, make sense! I'll update this. Cheers!